File: src/main/java/com/anod/appwatcher/installed/ImportAppViewHolder.java
package com.anod.appwatcher.installed;

import android.graphics.Color;
import android.view.View;
import android.widget.CheckedTextView;
import android.widget.ImageView;

import com.anod.appwatcher.R;
import com.anod.appwatcher.adapters.AppViewHolderBase;
import com.anod.appwatcher.model.AppInfo;
import com.anod.appwatcher.utils.AppIconLoader;

import butterknife.BindColor;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author alex
 * @date 2015-08-31
 */
class ImportAppViewHolder extends AppViewHolderBase implements View.OnClickListener {
    private final ImportDataProvider mDataProvider;

    public AppInfo app;
    @BindView(android.R.id.title)
    public CheckedTextView title;
    @BindView(android.R.id.icon)
    public ImageView icon;

    @BindColor(R.color.theme_accent)
    int themeAccent;
    @BindColor(R.color.material_red_800)
    int materialRed;

    ImportAppViewHolder(View itemView, ImportDataProvider dataProvider, AppIconLoader iconLoader) {
        super(itemView, dataProvider, iconLoader);
        mDataProvider = dataProvider;
        ButterKnife.bind(this, itemView);
    }

    public void bindView(int position, AppInfo app) {
        this.app = app;
        this.title.setText(app.title);
        boolean checked = mDataProvider.isPackageSelected(app.packageName);
        this.title.setChecked(checked);
        title.setEnabled(!mDataProvider.isImportStarted());

        itemView.findViewById(android.R.id.content).setOnClickListener(this);

        mIconLoader.loadAppIntoImageView(app, this.icon, R.drawable.ic_android_black_24dp);

        if (status() == ImportDataProvider.STATUS_DONE) {
            itemView.setBackgroundColor(themeAccent);
        } else if (status() == ImportDataProvider.STATUS_ERROR) {
            itemView.setBackgroundColor(materialRed);
        } else {
            itemView.setBackgroundColor(Color.TRANSPARENT);
        }
    }

    public int status() {
        return mDataProvider.getPackageStatus(app.packageName);
    }

    @Override
    public void onClick(View v) {
        this.title.toggle();
        mDataProvider.selectPackage(this.app.packageName, title.isChecked());
    }
}

File: app/src/main/java/com/anod/appwatcher/installed/ImportAppViewHolder.kt
package com.anod.appwatcher.installed

import android.graphics.Color
import android.view.View
import android.widget.CheckedTextView
import android.widget.ImageView
import androidx.core.content.res.ResourcesCompat
import com.anod.appwatcher.R
import com.anod.appwatcher.database.entities.AppListItem
import com.anod.appwatcher.utils.PicassoAppIcon
import com.anod.appwatcher.watchlist.AppViewHolderBase

/**
 * @author alex
 * *
 * @date 2015-08-31
 */
internal class ImportAppViewHolder(
        itemView: View,
        dataProvider: ImportResourceProvider,
        iconLoader: PicassoAppIcon)
    : AppViewHolderBase<AppListItem>(itemView, dataProvider, iconLoader), View.OnClickListener {

    private val importDataProvider = dataProvider
    private var item: AppListItem? = null
    private val title: CheckedTextView = itemView.findViewById(R.id.title)
    private val icon: ImageView = itemView.findViewById(R.id.icon)

    val themeAccent: Int = ResourcesCompat.getColor(itemView.resources, R.color.theme_accent, null)
    val rowError: Int = ResourcesCompat.getColor(itemView.resources, R.color.row_error_red, null)

    override fun bind(item: AppListItem) {
        this.item = item
        this.title.text = item.app.title
        val checked = importDataProvider.isPackageSelected(item.app.packageName)
        this.title.isChecked = checked
        title.isEnabled = !importDataProvider.isImportStarted

        itemView.findViewById<View>(R.id.content).setOnClickListener(this)

        iconLoader.loadAppIntoImageView(item.app, this.icon, R.drawable.ic_notifications_black_24dp)

        when {
            status() == ImportResourceProvider.STATUS_DONE -> itemView.setBackgroundColor(themeAccent)
            status() == ImportResourceProvider.STATUS_ERROR -> itemView.setBackgroundColor(rowError)
            else -> itemView.setBackgroundColor(Color.TRANSPARENT)
        }
    }

    override fun placeholder() {
        TODO("Not yet implemented")
    }

    fun status(): Int {
        return importDataProvider.getPackageStatus(item!!.app.packageName)
    }

    override fun onClick(v: View) {
        this.title.toggle()
        importDataProvider.selectPackage(item!!.app.packageName, title.isChecked)
    }
}